### PR TITLE
docs: clarify difference between kernel stub and sd-stub in UEFI doc

### PR DESCRIPTION
**Pull request by bluca**
For those who are unfamiliar with the project, it is not immediately clear from the document what's the story w.r.t. sd-stub vs kernel stub, so clarify it

### CODE REVIEW

File: src/boot/efi/UEFI_SECURITY.md [hunk 1]
```diff
@@ -4,6 +4,13 @@ PE binary, adding various features, `systemd-stub`. These components fully suppo
 this document will describe their security posture and how they comply with industry-standard expectations
 for UEFI SecureBoot workflows.
 
+Note that `systemd-stub` is not the same, or an alternative, to the Linux kernel's own EFI stub. The kernel
+stub's role is to prepare to launch the 'proper' kernel execution from EFI. `systemd-stub` on the other hand
```
Comment (poettering) [inline src/boot/efi/UEFI_SECURITY.md]: I think we should probably use the words "entrypoint" and "boot protocol" here.

Maybe:

The kernel stub's role is that of the fundamental entrypoint to kernel execution from UEFI mode, implementing the modern Linux boot protocol.

File: src/boot/efi/UEFI_SECURITY.md [hunk 2]
```diff
@@ -4,6 +4,13 @@ PE binary, adding various features, `systemd-stub`. These components fully suppo
...
+loads various resources, including the kernel image, via the EFI LoadImage/StartImage protocol (although it
+does support the legacy handover protocol, as a fallback for older kernels on x86). The purpose of
```
Comment (poettering) [inline src/boot/efi/UEFI_SECURITY.md]: maybe say "legacy Linux boot protocol", to contrast it nicely with the "modern Linux boot protocol"

File: src/boot/efi/UEFI_SECURITY.md [hunk 3]
```diff
@@ -4,6 +4,13 @@ PE binary, adding various features, `systemd-stub`. These components fully suppo
...
+`systemd-stub` is to provide additional features and functionality for both `systemd-boot` and `systemd` in
+userspace.
```
Comment (poettering) [inline src/boot/efi/UEFI_SECURITY.md]: `systemd-stub` offers nothing really for `systemd-boot`.


Maybe say: The purpose of `systemd-stub` is to provide additional features and functionality facing `systemd` in the OS userspace.

Comment (bluca) [reply src/boot/efi/UEFI_SECURITY.md]: it takes stuff passed from sd-boot via its protocol/variables, like kernel command line, so I'd say it is accurate that its purpose is to work with both components

Comment (poettering) [reply src/boot/efi/UEFI_SECURITY.md]: there's nothing sd-boot specific in there.

Comment (bluca) [reply src/boot/efi/UEFI_SECURITY.md]: Ok, Boot Loader Interface specific then? This is an explanatory paragraph added after direct feedback that things weren't clear, so I am more concerned in ensuring a non-expert reader gets the right kind of idea of the split and relationship between these components, than being strictly adherent to the letter of the definitions (in this paragraph), but if you want to improve the wording please feel free to send a follow-up

**Pull request merged**